File: src/main/scala/net/liftweb/seventhings/snippet/FetchAd.scala
package net.liftweb.seventhings
package snippet

import net.liftweb._
import util._
import Helpers._

/**
 * Fetch an ad from an ad server (takes about 1/2 second
 */
object FetchAd {
  def render = {
    // sleep for 1/2 second
    Thread.sleep(500 millis)

    // send the result back
    ".ad" #> Thread.currentThread.getName
  }
}

File: src/main/java/net/liftweb/seventhings/snippet/FetchAd.java
package net.liftweb.seventhings.snippet;

import net.liftweb.util.Css;
import scala.Function1;
import scala.xml.NodeSeq;

public class FetchAd {

    // return the code that transforms the template input to
    // the dynamic output
    public Function1<NodeSeq, NodeSeq> render() {

        // sleep for 1/2 second
        try {
            Thread.sleep(500);
        } catch (InterruptedException IGNORE) {}

        // send the result back
        return Css.sel(".ad", Thread.currentThread().getName());

    }


}
